symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -63,8 +64,96 @@ void test_cwrapper() {
     basic_str_free(s);
 }
 
+void test_CVectorInt1()
+{
+    // Allocate on heap
+    CVectorInt *vec = vectorint_new();
+    vectorint_push_back(vec, 5);
+    assert(vectorint_get(vec, 0) == 5);
+    vectorint_free(vec);
+}
+
+struct X {
+    void *x;
+};
+
+void test_CVectorInt2()
+{
+    // Allocate on stack
+    CVectorInt *vec;
+
+    char data1[1];  // Not aligned properly
+    vec = (CVectorInt*)data1;
+    assert(vectorint_placement_new(vec, sizeof(data1)) == 2);
+
+    struct X data2[1];  // Aligned properly but small
+    vec = (CVectorInt*)data2;
+    assert(vectorint_placement_new(vec, sizeof(data2)) == 1);
+
+    char data3[50]; // Aligned properly and enough size to fit std::vector<int>
```

Comment:
An array is only required to have the alignment of a member, which is 1 byte in this case (alignment of char). For example,
```
    char d[50];
    std::cout << alignof(decltype(d)) << std::endl;
```
gives me 1